### DISCUSSION TITLE
Support for static hasMany, static belongsTo in domain

### DISCUSSION THREAD

**Discussion by sk-space** [micronaut-projects/micronaut-data#1028] [Q&A]
Does [Micronaut Data](https://micronaut-projects.github.io/micronaut-data/latest/guide/) (micronaut-hibernate-gorm) support `static hasMany`, `static belongsTo` in domain class to address entity mapping/associations as provided by [GORM](http://gorm.grails.org/latest/hibernate/manual/) in [GRAILS](https://docs.grails.org/4.0.10/guide/single.html)?

When I tried, when retrieving data org.hibernate.LazyInitilizationExcepiton occured on many side of relation.

Parent Domain class:
```
@CompileStatic
@Entity
class Parent implements GormEntity<Parent> {

    UUID code
    String name
    Double totalCapacity

    static hasMany = [children: Child]

    // static fetchMode = [children: 'lazy']

    static constraints = {

    }
}
```

Child Domain class:
```
@CompileStatic
@Entity
class Child implements GormEntity<Child> {

    UUID code
    String name

    static belongsTo = [parent: Parent]

    static constraints = {

    }

}
```

Exception message:
1. Unable to evaluate the expression Method threw 'org.hibernate.LazyInitializationException' exception.
2. Caused by: org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: com.project.domain.component.Tank.nozzels, could not initialize proxy - no Session

PS: Please ignore the typos.

**Comment by dstepanov** [comment]
You might need transactional annotation somewhere. Maybe you can provide a project with the problem to tell more.

**Reply by sk-space** [reply]
Please check out from this repo: [DEMO](https://github.com/sk-space/demo-micronaut)
